privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/animations/LottiePrivacyShieldView.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.privacy.dashboard.impl.animations
+
+import com.airbnb.lottie.LottieAnimationView
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.ui.store.AppTheme
+import com.duckduckgo.privacy.dashboard.api.PrivacyShield
+import com.duckduckgo.privacy.dashboard.api.PrivacyShield.PROTECTED
+import com.duckduckgo.privacy.dashboard.api.PrivacyShield.UNKNOWN
+import com.duckduckgo.privacy.dashboard.api.PrivacyShield.UNPROTECTED
+import com.duckduckgo.privacy.dashboard.api.PrivacyShield.WARNING
+import com.duckduckgo.privacy.dashboard.api.animations.PrivacyShieldView
+import com.duckduckgo.privacy.dashboard.impl.R
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import timber.log.Timber
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
```

Comment:
Do we want this to be a singleton? Don't have the full context but either it should not be a Singleton or the name is not great because I assume this is a view and is probably not.

Reply:
It can be a singleton because it's stateless and we don't produce effects to any external component. But it's up to us. I would say the name is not great.

Going to change to something like `LottiePrivacyShieldAnimationHelper`, let me know if that's fine with you.

Reply:
👍